### ISSUE TITLE
Debug JS Remotely

### ISSUE THREAD

**Issue by sm2017** [facebook/react-native#15805]
### Is this a bug report?

(Yes , 100000000%)

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

(Yes)

### Environment

1. `react-native -v`:
```
react-native-cli: 2.0.1
react-native: 0.47.1
```

2. `node -v`:
```
v8.1.3
```

3. `npm -v`:
```
5.0.3
```

4. `yarn --version`:
```
0.27.5
```

Then, specify:

- Target Platform: Android

- Development Operating System: Windows10

- Build tools: `react-native run-android`

### Steps to Reproduce

You cannot easily reproduce error , this is a random cryptic error

### Expected Behavior

Executing JS code must not depends on remote debug , So works well when remote debug is off or on

### Actual Behavior

 For android when remote debug is ON , code works and when OFF not works 

Foe more information see:

https://medium.com/@jordan_mohi/cant-find-variable-symbol-in-react-native-782489d86bd7
https://github.com/facebook/react-native/issues/15786
https://github.com/facebook/react-native/issues/15497

### Reproducible Demo

(N/A)

**Comment by iegik** [comment]
I think, it's not a bug, it is a feature.
With Debug enabled, I think for debug support included some additional es6 modules, such as Request, Symbol or some of RegExp features, So, ensure, that You have all dependences included, before start to scripting.

For example,
```
npm install --save es6-symbol
import 'es6-symbol/implement';
```

In production mode, you would not need to have unneeded script included.

**Comment by sm2017** [comment]
@iegik I think it is a bug because , I use `Symbol` in a js file from 6 months ago , I don't change this file from   months ago , suddenly react-native `Can’t find variable: Symbol` only on android when remote debug is not enabled (see https://github.com/facebook/react-native/issues/15786). If it is not a bug and a feature why iOS has not this feature?

If you think it is a feature , Why we lost BackHandler on android when remote debug is off? (See https://github.com/facebook/react-native/issues/15497)

**Comment by iegik** [comment]
@sm2017 You can also try to debug with [Android Monitor](https://developer.android.com/studio/profile/android-monitor.html) or [logcat](https://developer.android.com/studio/command-line/logcat.html)

**Comment by dummerbd** [comment]
Just fyi for people finding this issue via Google, the problem as I understand it is `Symbol` is not supported in the current version of the JavaScript Core shipped with Android but it is supported natively on iOS which has a newer version of JSC. The reason you don't see this issue when using Remote Debug is because your JavaScript actually runs remotely in Chrome which does support `Symbol`.

Long story short, use a polyfill for now like `es6-symbol`.

**Comment by joeytwiddle** [comment]
A possible polyfill solution is [mentioned here](https://github.com/facebook/react-native/issues/4676#issuecomment-340290485)
